src/commands/external.rs

```diff
@@ -13,15 +14,65 @@ pub async fn execute_external_subcommand(args: Vec<String>) -> Result<()> {
     let plugin_name = args.first().ok_or_else(|| anyhow!("Expected subcommand"))?;
     let plugins_dir = get_spin_plugins_directory()?;
     check_plugin_spin_compatibility(plugin_name, env!("VERGEN_BUILD_SEMVER"), &plugins_dir)?;
-    let path = plugins_dir.join(plugin_name).join(plugin_name);
-    let mut command = Command::new(path);
+    let path = plugins_dir.join(plugin_name);
+    let binary = path.join(plugin_name);
+    let mut command = Command::new(binary);
     if args.len() > 1 {
         command.args(&args[1..]);
     }
+    command.envs(&get_env_vars_map(&path)?);
     log::info!("Executing command {:?}", command);
     // Allow user to interact with stdio/stdout of child process
     let _ = command.status().await?;
     // TODO: handle the status
 
     Ok(())
 }
+
+fn get_env_vars_map(path: &Path) -> Result<HashMap<String, String>> {
+    let map: HashMap<String, String> = vec![
+        (
+            "SPIN_VERSION".to_string(),
+            env!("VERGEN_BUILD_SEMVER").to_owned(),
+        ),
+        (
+            "SPIN_BIN_PATH".to_string(),
+            env::current_exe()?.to_str().unwrap_or_default().to_owned(),
+        ),
+        (
+            "SPIN_PLUGIN_PATH".to_string(),
+            path.to_str().unwrap_or_default().to_owned(),
```

Comment:
shouldnt we error if the bin or plugin path DNE? `ok_or_else(|| anyhow!("Could not convert <<bin or plugin>> path to string"))?`